### DISCUSSION TITLE
Saved lora hashes get repeated from previous image

### DISCUSSION THREAD

**Discussion by acorderob** [vladmandic/sdnext#2522] [General]
I feel like this should be open as an issue, but i haven't tested it without extensions, so i want to see if someone else reproduces it.

When generating multiple consecutive images with batch count, and using wildcards so the prompt sometimes have loras and sometimes don't, an image with no loras will get metadata saved with the lora hashes of the previously generated image (that had loras). If the image has loras the metadata will be correct. Not sure if this "caching" is repeated over multiple images, but probably is.

Seems like that the metadata is missing an initialization on each image or it is in the wrong place.

Does anyone else reproduce this?

I'm using commit 7cbd2bc9.

**Comment by vladmandic** [comment]
model/network initialization is done once-per-batch and although changing network params inside batch may be doable via use of extensions, that is not what batch is intended for. if you do so, you do it at your own risk.

**Reply by acorderob** [reply]
I don't quite understand what you are saying. The model+loras are initialized and used correctly in each image. It's the saving of the used loras what's wrong in the metadata, and only when the image does not use any loras.
